syntax = "proto3";

package transport_catalogue_proto;
 
message Rgb {
    uint32 r = 1;
    uint32 g = 2;
    uint32 b = 3;
}
 
message Rgba {
    uint32 r = 1;
    uint32 g = 2;
    uint32 b = 3;
    double o = 4;
}
 
message Color {
    oneof color {
        Rgb rgb = 1;
        Rgba rgba = 2;
        string string_color = 3;
        bool empty = 4;
    }
}
